contracts/truefi2/TrueFiCreditOracle.sol

```diff
@@ -75,9 +103,9 @@ contract TrueFiCreditOracle is ITrueFiCreditOracle, UpgradeableClaimable {
      * Scores are stored as uint8 allowing scores of 0-255
      */
     function setScore(address account, uint8 newScore) public onlyManager {
+        _setEligibleUntilTime(account, Math.max(eligibleUntilTime[account], block.timestamp.add(creditUpdatePeriod)));
```

Comment:
👍 Nice that You handled the case when we want to set eligibility for a longer than standard period